test/integration/header_integration_test.cc

```diff
@@ -333,8 +341,10 @@ class HeaderIntegrationTest : public HttpIntegrationTest,
                       Http::TestHeaderMapImpl& expected_headers) {
     headers.remove(Envoy::Http::LowerCaseString{"content-length"});
     headers.remove(Envoy::Http::LowerCaseString{"date"});
-    headers.remove(Envoy::Http::LowerCaseString{"x-envoy-expected-rq-timeout-ms"});
-    headers.remove(Envoy::Http::LowerCaseString{"x-envoy-upstream-service-time"});
+    if (!routerSuppressEnvoyHeaders()) {
```

Comment:
headers.removePrefix("x-envoy-") for future-proofing?

Reply:
Some of the `x-envoy-` are not added by the router, so it's not possible to do the prefix removal unfortunately.